component-tools/src/main/java/org/talend/sdk/component/tools/ComponentValidator.java

```diff
@@ -1161,6 +1168,45 @@ private void validateActions(final AnnotationFinder finder, final Set<String> er
                         .collect(toSet()));
     }
 
+    private void validateExceptions(final Set<String> errors) {
+        System.out.println("Validate exceptions");
+
+        boolean exceptionFound = Arrays
+                .stream(classes)
+                .flatMap(f -> streamClassesInDirectory(null, f))
+                .filter(ComponentException.class::isAssignableFrom)
+                .findFirst()
+                .isPresent();
+
+        if (!exceptionFound) {
+            errors.add("Component should declare a custom ComponentException;");
+        }
+    }
+
+    private Stream<Class> streamClassesInDirectory(final String pckg, final File classFile) {
+
+        System.out.println("Looking into " + classFile);
+
+        if (classFile.isDirectory()) {
+            return Arrays
+                    .stream(classFile.listFiles())
+                    .flatMap(f -> streamClassesInDirectory(pckg == null ? "" : (pckg + classFile.getName() + "."), f));
+        }
+
+        ClassLoader loader = Thread.currentThread().getContextClassLoader();
+        if (classFile.getName().endsWith(".class")) {
+            String className = classFile.getName().substring(0, classFile.getName().lastIndexOf("."));
+            try {
+                System.out.println("Loading class " + pckg + className);
+                return Stream.of(loader.loadClass(pckg + className));
+            } catch (Exception e) {
+                log.info("Could not load class : " + pckg + className + "=>" + e.getMessage());
```

Comment:
Here, you use at least a logger but with an info level on an exception catch...
Use warning level instead.

Reply:
no warning level in log, I'll use error